mars/utils.py

```diff
@@ -519,3 +524,54 @@ def build_fuse_chunk(fused_chunks, **kwargs):
                                         _operands=[c.op for c in fused_chunks])
     return fuse_op.new_chunk(head_chunk.inputs, kws=[params], _key=tail_chunk.key,
                              _composed=fused_chunks, **kwargs)
+
+
+def merge_chunks(chunk_results):
+    """
+    Concatenate chunk results according to index.
+    :param chunk_results: list of tuple, {(chunk_idx, chunk_result), ...,}
+    :return:
+    """
+    from .lib.sparse import SparseNDArray
+
+    chunk_results = sorted(chunk_results, key=lambda x: x[0])
+    v = chunk_results[0][1]
+    if isinstance(v, (np.ndarray, SparseNDArray)):
+        ndim = v.ndim
+        for i in range(ndim - 1):
+            new_chunks = []
+            for idx, cs in itertools.groupby(chunk_results, key=lambda t: t[0][:-1]):
+                new_chunks.append((idx, np.concatenate([c[1] for c in cs], axis=ndim - i - 1)))
```

Comment:
For a `SparseNDArray`, does `numpy.concatenate` trigger `__array__` protocol and lead to memory explosion?

Reply:
Oh, I will fix it.